InfiniteRecharge/src/main/java/frc/robot/subsystem/scoring/shooter/ShooterSubsystem.java

```diff
@@ -184,15 +188,23 @@ public void periodic(float deltaTime) {
     }
 
     public void spinUp() {
+        float targetShooterVelocity = (float) MathUtils
+                .unitConverter(SmartDashboard.getNumber(getName() + "/Shooter Velocity RPM", 60), 600, 8192)
+                / config.shooter.shooterGearRatio;
+
         // Spin up the feeder.
-        feeder.set(
-                SmartDashboard.getNumber(getName() + "/Feeder Output Percent", ShooterConstants.FEEDER_OUTPUT_PERCENT));
-        SmartDashboard.putString(getName() + "/Feeder State", "Feeding");
+        if (ballPropulsionMotor.getSelectedSensorVelocity() >= targetShooterVelocity
+                - config.shooter.feederSpinUpDeadband
+                && ballPropulsionMotor.getSelectedSensorVelocity() <= targetShooterVelocity
+                        + config.shooter.feederSpinUpDeadband) {
+            feeder.set(SmartDashboard.getNumber(getName() + "/Feeder Output Percent",
+                    ShooterConstants.FEEDER_OUTPUT_PERCENT));
+            SmartDashboard.putString(getName() + "/Feeder State", "Feeding");
+            upToSpeed = true;

```

Comment:
This is never set to false if the user is holding down the button and the shooter drops in velocity because it shot a few balls out.